### ISSUE TITLE
stubborn merge segment after a singleton operator

### ISSUE THREAD

**Issue by tconbeer** [tconbeer/sqlfmt#203]
**Describe the bug**
If an operator ends up on a line by itself (after splitting), we don't try hard enough to merge the segment after that operator into the operator's line, which can result in ugly output (for somewhat pathological input)

**To Reproduce**
From gitlab's project:
```sql
{{column|lower}} || 
ENCRYPT_RAW(
    to_binary('{{ get_salt(column|lower) }}', 'utf-8'), 
    to_binary('{{ env_var("SALT_PASSWORD") }}', 'HEX'), 
    to_binary('416C736F4E637265FFFFFFAB', 'HEX')
)
```

**Expected behavior**
```sql
{{ column | lower }} || encrypt_raw(
    to_binary('{{ get_salt(column|lower) }}', 'utf-8'),
    to_binary('{{ env_var("SALT_PASSWORD") }}', 'HEX'),
    to_binary('416C736F4E637265FFFFFFAB', 'HEX')
)
```

**Actual behavior**
```sql
{{ column | lower }} ||
encrypt_raw(
    to_binary('{{ get_salt(column|lower) }}', 'utf-8'),
    to_binary('{{ env_var("SALT_PASSWORD") }}', 'HEX'),
    to_binary('416C736F4E637265FFFFFFAB', 'HEX')
)
```

**Additional context**
What is the output of `sqlfmt --version`?
Commit 88617d2 (v0.10)

**Comment by tconbeer** [comment]
See also:

https://github.com/tconbeer/gitlab-analytics-sqlfmt/blob/7e002680565326278d4ead72cff9c651f27c7169/transform/snowflake-dbt/models/legacy/bamboohr/rpt/bamboohr_rpt_headcount_aggregation.sql#L88-L89

**Comment by tconbeer** [comment]
Also need to handle open parens that end up on lines by themselves, e.g., `over\n(something long)`

https://github.com/tconbeer/gitlab-analytics-sqlfmt/blob/7e002680565326278d4ead72cff9c651f27c7169/transform/snowflake-dbt/models/legacy/bamboohr/sensitive/bamboohr_job_role.sql#L42-L44

**Comment by tconbeer** [comment]
Also:

https://github.com/tconbeer/rittman_ra_data_warehouse/blob/d8dc7bd1c008ca79f9d09c909734e28a66ef6366/models/integration/int_ad_campaign_performance.sql#L4-L5

**Comment by tconbeer** [comment]
closed by [fd66993](https://github.com/tconbeer/sqlfmt/pull/210/commits/fd669933c5a4bf6981bbc3446fb419757fc49fcb)

**Comment by tconbeer** [comment]
closed in #210
